src/consensus/pbft/pbft.h

```diff
@@ -71,11 +82,76 @@ namespace pbft
         reinterpret_cast<const uint8_t*>(msg->contents()),
         msg->size());
 
+      if (msg->tag() == Append_entries_tag)
+      {
+        auto node = nodes.find(to);
+
+        Index match_idx = 0;
+        if (node != nodes.end())
+        {
+          match_idx = node->second;
+        }
+
+        if (match_idx < latest_stable_ae_index)
+        {
+          send_append_entries(to, match_idx + 1);
+        }
+        return msg->size();
+      }
       n2n_channels->send_authenticated(
         ccf::NodeMsgType::consensus_msg, to, serialized_msg);
       return msg->size();
     }
 
+    void send_append_entries(NodeId to, Index start_idx)
+    {
+      size_t entries_batch_size = 10;
+
+      Index end_idx = (latest_stable_ae_index == 0) ?
+        0 :
+        std::min(start_idx + entries_batch_size, latest_stable_ae_index);
+
+      for (Index i = end_idx; i < latest_stable_ae_index;
+           i += entries_batch_size)
+      {
+        send_append_entries_range(to, start_idx, i);
+        start_idx = std::min(i + 1, latest_stable_ae_index);
+      }
+
+      if (latest_stable_ae_index == 0 || end_idx <= latest_stable_ae_index)
+      {
+        send_append_entries_range(to, start_idx, latest_stable_ae_index);
+      }
+    }
+
+    void send_append_entries_range(NodeId to, Index start_idx, Index end_idx)
+    {
+      const auto prev_idx = start_idx - 1;
+
+      LOG_INFO_FMT(
+        "Send append entried from {} to {}: {} to {}",
+        id,
+        to,
+        start_idx,
+        end_idx);
+
+      AppendEntries ae = {pbft_append_entries, id, end_idx, prev_idx};
+
+      auto node = nodes.find(to);
+      if (node != nodes.end())
+      {
+        node->second = end_idx;
+      }
+      else
+      {
+        nodes[to] = end_idx;
+      }
+
+      // The host will append log entries to this message when it is
```

Comment:
This is really cool because it means we can now ship the same `cchost` for both Raft and PBFT.